### ISSUE TITLE
Calling app.getDocumentStructure() in pulgin crash.

### ISSUE THREAD

**Issue by Nistek** [xournalpp/xournalpp#4192]
**Affects versions :**
 - OS: Windows 7 Pro 64 bits
 - Which version of libgtk do you use: 3.24.34
 - Version of Xournal++:  1.1.1+dev, 7b6d8495-dirty, Aug 5 2022
 - Installation method: Windows installer

**Describe the bug**
In a document with one page and 3 layers and a second page with only one layer with this last one selected, then i call to the function app.getDocumentStructure() from a pulgin , Xournal++ crashed.

**To Reproduce**
1. Download Debug  pulgin and put on Xournal++ pulgin directory(<Xournal++ root>\share\xournalpp\plugins).
2. Open Xournal++ and activate Debug pulgin.
3. Restart Xournal++
4. Create 2 layers.
5. Create new page.
6. Goto Pulgin->Print pointer or Pulgin->Print Document Structure
7. Crash

**Expected behavior**
Do not crash, instead of that appear a dialog box with info abut document structure.

**Screenshots of Problem**
See video below...

**Additional context**
Here include the pulgin: [Debug.zip](https://github.com/xournalpp/xournalpp/files/9294942/Debug.zip)
Although i think that in any plugin that is called under that state Xournal++ crash.

https://user-images.githubusercontent.com/61591806/183774416-8aec597a-c4f3-4eb0-adde-59fa5af76aa7.mp4

**Comment by rolandlo** [comment]
Thanks for reporting the bug. I could reproduce it. The errorlog shows that the problem is related to a segfault when trying to access the layer name:

```
Error: signal 11
[bt]: (0) xournalpp(+0xafc88) [0x55c10967fc88]
[bt]: (1) /lib/x86_64-linux-gnu/libc.so.6(+0x42520) [0x7fe7ace42520]
[bt]: (2) xournalpp(_ZNK15LayerController19getCurrentLayerNameB5cxx11Ev+0xb0) [0x55c1097223b0]
[bt]: (3) xournalpp(_ZNK15LayerController16getLayerNameByIdB5cxx11Em+0xe4) [0x55c1097225a4]
```

**Comment by rolandlo** [comment]
A fix is available in PR #4196. This will be first merged into the release-1.1 branch and only later to the master branch.
